### ISSUE TITLE
Quotes and parentheses in SQL query

### ISSUE THREAD

**Issue by montekidlo** [orientechnologies/orientdb#6786]
## OrientDB Version, operating system, or hardware.
- v2.2.5
## Operating System
- Linux
## Expected behavior and actual behavior

I have next data:

```
{
"result": [
   {
       "@type": "d",
       "@rid": "#25:0",
       "@version": 1,
       "@class": "name",
       "surname": "foo surname",
       "name": "foo name"
   },
   {
       "@type": "d",
       "@rid": "#26:0",
       "@version": 1,
       "@class": "name",
       "surname": "bar surname",
       "name": "bar name"
   },
   {
       "@type": "d",
       "@rid": "#27:0",
       "@version": 1,
       "@class": "name",
       "surname": "O'brien",
       "name": "baz"
   }
],
"notification": "Query executed in 0.015 sec. Returned 3 record(s)"
}
```

for query

```
SELECT * FROM name;
```

And I have next query:

```
select * from name WHERE (NOT (surname='O\'brien'))
```

And result:

```
java.lang.NullPointerException
```

Expected result:

```
{
"result": [
   {
       "@type": "d",
       "@rid": "#25:0",
       "@version": 1,
       "@class": "name",
       "surname": "foo surname",
       "name": "foo name"
   },
   {
       "@type": "d",
       "@rid": "#26:0",
       "@version": 1,
       "@class": "name",
       "surname": "bar surname",
       "name": "bar name"
   }
]
}
```

More examples in http://stackoverflow.com/questions/39892358/orientdb-quote-in-sql-query

Question - is it a bug, or I do something wrong?

Thanks.

**Comment by luigidellaquila** [comment]
Hi @montekidlo 

it seems to be a bug, I'm on it

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @montekidlo 

I just pushed a fix in 2.2.x branch. The fix will be released with 2.2.13

Thanks

Luigi
